File: Models/Expense.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MilaAPI.Models
{
	public class Expense
	{
		public int Id { get; set; }

		[Column(TypeName = "decimal(18,2)")] // This ensures the SQL column is created with the correct precision and scale
		public decimal Amount { get; set; }
		public DateTime Date { get; set; } // Date the expense was made
		public string Description { get; set; } // Optional description of the expense

		// Foreign key to the User entity
		public int UserId { get; set; }
		public User User { get; set; } // Navigation property

		// Foreign key to Category entity
		public int CategoryId { get; set; }
		public Category Category { get; set; } // Navigation property
	}
}

File: front-end/milaapp/lib/models/expense.dart
class Expense {
  final String id;
  final double amount;
  final DateTime date;
  final String description;
  final String categoryId;
  final String userId;

  Expense({
    required this.id,
    required this.amount,
    required this.date,
    required this.description,
    required this.categoryId,
    required this.userId,
  });

  factory Expense.fromJson(Map<String, dynamic> json) {
    return Expense(
      id: json['id'],
      amount: json['amount'],
      date: DateTime.parse(json['date']),
      description: json['description'],
      categoryId: json['categoryId'],
      userId: json['userId'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'amount': amount,
      'date': date.toIso8601String(),
      'description': description,
      'categoryId': categoryId,
      'userId': userId,
    };
  }
}
